### DISCUSSION TITLE
whether Sparta support build accurate time model with systemc?

### DISCUSSION THREAD

**Discussion by dzwduan** [sparcians/map#400] [General]

**Comment by ghost** [comment]
Not sure I follow your question...

If you're looking at integrating a SystemC model with Sparta, that's certainly possible and has been done before.  

First, documentation: https://sparcians.github.io/map/systemc.html
Second, examples using TLM: https://github.com/sparcians/map/tree/master/sparta/example/SystemC

On the documentation page, you'll find instructions on how to build Sparta with SystemC support.  Then, build and run the SystemC examples to understand how the frameworks interact.   

One caveat to note: 0-cycle communication between a Sparta unit and a SystemC component does not work if the user is expecting a Sparta unit to schedule an event and provide a response _in the same cycle_ to the SystemC component.

**Reply by dzwduan** [reply]
Thanks a lot, i will try to build riscv cycle accurate model only based on sparta
